mapie/calibration.py

```diff
@@ -0,0 +1,502 @@
+        calibrator = self._get_calibrator(cv, self.calibrator)
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        assert type_of_target(y) in ["multiclass", "binary"]
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+        self.n_features_in_ = check_n_features_in(X, cv, estimator)
+        random_state = check_random_state(random_state)
+
+        if cv == "prefit":
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.n_classes_ = len(self.classes_)
+            self.calibrators = self._fit_calibrators(
+                X, y, sample_weight, estimator, calibrator
+            )
+        if cv == "split":
+            results = check_calib_set(
+                X,
+                y,
+                sample_weight=sample_weight,
+                X_calib=X_calib,
+                y_calib=y_calib,
+                sample_weight_calib=cast(
+                    Optional[NDArray],
+                    sample_weight_calib
+                ),
+                calib_size=calib_size,
+                random_state=random_state,
+                shuffle=shuffle,
+                stratify=stratify,
+            )
+            X_train, y_train, X_calib, y_calib, sw_train, sw_calib = results
+            X_train, y_train = indexable(X_train, y_train)
+            y_train = _check_y(y_train)
+            sw_train, X_train, y_train = check_null_weight(
+                sw_train,
+                X_train,
+                y_train
+            )
+            self.n_classes_ = len(np.unique(y_train))
+            self.estimator = fit_estimator(
+                clone(estimator), X_train, y_train, sw_train,
+            )
+            self.classes_ = self.estimator.classes_  # type: ignore
+            self.calibrators = self._fit_calibrators(
+                X_calib, y_calib, sw_calib, self.estimator, calibrator
+            )
+        return self
+
+    def predict_proba(
+        self,
+        X: ArrayLike,
+    ) -> NDArray:
+        """
+        Prediction of the calibrated probability score of the class after
+        fitting of the classifer and calibrator.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        Returns
+        -------
+        NDArray of shape (n_samples, n_classes)
+            The calibrated score for each max score and zeros at every
+            other position in that line.
+        """
+        check_is_fitted(self, self.fit_attributes)
+        y_pred_probs = self.estimator.predict_proba(X=X)  # type: ignore
+        # y_pred_probs = fix_number_of_classes(
+        #     self.n_classes_,
+        #     self.estimator.classes_,  # type: ignore
+        #     y_pred_probs
+        # )
+        self.uncalib_pred = y_pred_probs
+
+        max_prob, max_prob_arg = self._get_labels(
+            cast(str, self.method),
+            y_pred_probs
+        )
```

Comment:
same remark, rename max_prob_arg to y_pred

Reply:
done.